client/blocks/editor-checkout-modal/index.tsx

```diff
@@ -82,29 +82,31 @@ const EditorCheckoutModal = ( props: Props ) => {
 		: cartData.products.map( ( product ) => product.product_slug );
 	const commaSeparatedProductSlugs = productSlugs?.join( ',' ) || null;
 
-	return hasEmptyCart ? null : (
-		<Modal
-			open={ isOpen }
-			overlayClassName="editor-checkout-modal"
-			onRequestClose={ onClose }
-			title=""
-			shouldCloseOnClickOutside={ false }
-			icon={ <Icon icon={ wordpress } size={ 36 } /> }
-		>
-			<ShoppingCartProvider cartKey={ cartKey } getCart={ wpcomGetCart } setCart={ wpcomSetCart }>
-				<StripeHookProvider
-					fetchStripeConfiguration={ fetchStripeConfigurationWpcom }
-					locale={ props.locale }
-				>
-					<CompositeCheckout
-						isInEditor
-						siteId={ site.ID }
-						siteSlug={ site.slug }
-						productAliasFromUrl={ commaSeparatedProductSlugs }
-					/>
-				</StripeHookProvider>
-			</ShoppingCartProvider>
-		</Modal>
+	return (
+		isOpen && (
```

Comment:
What is the motivation for changing the check from `hasEmptyCart` to `isOpen` ?

Reply:
`isOpen` was added in context of https://github.com/Automattic/wp-calypso/pull/47601#discussion_r529419721

I'll check this and probably re-add `hasEmptyCart` as an additional condition.

Reply:
`hasEmptyCart` is not actually checking if the cart is empty but rather if there are any products passed to the `EditorCheckoutModal` component. 

In the case of Launch, we're [just calling the hook](https://github.com/Automattic/wp-calypso/pull/47601/files#diff-2f5e6bd15fcaddc71644575361bf6a5f0193c51502cd8c5337a40d345d897cc9R41) to open the Checkout modal, without passing any products. They are already [added via a REST API call](https://github.com/Automattic/wp-calypso/blob/9644ddf7d0a588a2774d5b245664273a45079ab5/packages/launch/src/hooks/use-cart.ts#L30) to the cart before that.

Reply:
> `hasEmptyCart` is not actually checking if the cart is empty but rather if there are any products passed to the EditorCheckoutModal component.

Good to know

> In the case of Launch, we're just calling the hook to open the Checkout modal, without passing any products. They are already added via a REST API call to the cart before that.

Thank makes sense. We just need to make sure that changing the check from `hasEmptyCart` to `isOpen` won't affect any other flows that are using this block

Reply:
Tested premium blocks upgrade and looks fine. It's also E2E tested now 😁

Reply:
gotta love e2e tests!